src/replication.c

```diff
@@ -488,6 +497,57 @@ trigger_rebuild(spec_t *spec)
 	}
 }
 
+static bool
+is_replica_familiar(spec_t *spec, replica_t *new_replica)
+{
+	known_replica_t *kr = NULL;
+	int connected_replica = 0;
+	bool ret = false, found = false;
+	replica_t *old_replica = NULL;
+
+	ASSERT(MTX_LOCKED(&spec->rq_mtx));
+
+	TAILQ_FOREACH(kr, &spec->identified_replica, next) {
+		connected_replica++;
+		if (kr->zvol_guid == new_replica->zvol_guid) {
+			found = true;
+			if (!kr->is_connected) {
+				ret = true;
+			}
+			break;
+		}
+	}
+
+	if (!found && connected_replica < spec->replication_factor) {
+		kr = malloc(sizeof (known_replica_t));
+		kr->zvol_guid = new_replica->zvol_guid;
+		kr->is_connected = true;
```

Comment:
What if replica is already there in identified_replica list but not connected(is_connected == false) where would we set it to true when we move this replica to ready_queue (spec->rq) ? Connected_replica should be count of replicas which are actually connected i.e. kr->is_connected should be true for those replicas otherwise we are just counting stale replica.